WordPress/gutenberg
# Update theme.json: split global block selector into root and defaults

This PR continues the work started at https://github.com/WordPress/gutenberg/pull/28110 to update the shape of theme.json according to https://github.com/WordPress/gutenberg/issues/28163 It splits the existing `global` block selector in two:

- `root` => to address the site root
- `defaults` => to address all blocks at once

Before this change, the settings under the `global` block selector would have two functions: serve as default values for the other blocks and actually set the "global" block:

```
{
  "settings": {
    "global": { ... },
    "core/paragraph": { ... }
  },
  "styles": {
    "global": { ... },
    "core/paragraph": { ... }
  }
}
```

After this change, the default values should be declared under the `defaults` block selector and the `root` only contains settings & styles for the block that represents the root of the site:


```
{
  "settings": {
    "defaults": { ... }, // The settings here serve as defaults that any block can overwrite
    "root": { ... }, // Settings specific to the root block
    "core/paragraph": { ... } // Settings specific to the paragraph block
  },
  "styles": { // The styles section can't have the `*` selector.
    "root": { ... },
    "core/paragraph": { ... }
  }
}
```

See [updated documentation](https://github.com/WordPress/gutenberg/blob/f0681eb2394df26ec9928219ff75538781faea45/docs/designers-developers/developers/themes/theme-json.md) for more details.

## How to test


- Use this branch from TT1-blocks for testing WordPress/theme-experiments#175
- Make sure the front-end of the site looks the same.
- Go to the post editor and fiddle with it a bit: especially test that presets, custom CSS variables, and editor configuration work as expected.
- Go to the site editor and update some user configurations (background color at the global level, etc). Make sure the changes are reflected in the editor. Save user changes and go to the front-end: you should see the same changes.
- Change some settings & styles in the theme.json of the WordPress/theme-experiments#175 and make sure they work as expected (both front-end and editors).
